File: sokoban/mainpanel.py
from kivy.uix.gridlayout import GridLayout


class MainPanel:
    def __init__(self):
        self._game_panel = None
        self._create_view()
        self._init_view()

    def set_game_panel(self, _game_panel):
        self._remove_existing_game_panel()
        self._game_panel = _game_panel
        self.view.add_widget(self._game_panel)

    def _remove_existing_game_panel(self):
        if self._game_panel:
            self.view.remove_widget(self._game_panel)

    def _create_view(self):
        self.view = GridLayout()

    def _init_view(self):
        self.view.rows = 1
        self.view.cols = 1

File: web-sokoban/mainpanel.js
class MainPanel {
    constructor() {
        this._game_panel = null
        this._create_view()
        this._init_view()
    }

    set_game_panel(_game_panel) {
        this._remove_existing_game_panel()
        this._game_panel = _game_panel
        this.view.add_widget(this._game_panel)
    }

    _remove_existing_game_panel() {
        if (this._game_panel) {
            this.view.remove_widget(this._game_panel)
        }
    }

    _create_view() {
        this.view = GridLayout()
    }

    _init_view() {
        this.view.rows = 1
        this.view.cols = 1
    }
}
